migration/identity-migration/src/main/java/io/camunda/migration/identity/handler/sm/GroupMigrationHandler.java

```diff
@@ -0,0 +1,229 @@
+                targetGroupId);
+          }
+        });
+  }
+
+  private void createAuthorizationsForGroup(final String groupId, final String targetGroupId) {
+    final List<Authorization> authorizations;
+    try {
+      authorizations = managementIdentityClient.fetchGroupAuthorizations(groupId);
+    } catch (final Exception e) {
+      if (!isNotImplementedError(e)) {
+        throw new MigrationException(
+            String.format("Failed to fetch authorizations for group with ID '%s'", groupId), e);
+      }
+      logger.warn("Authorization endpoint is not available, skipping.");
+      return;
+    }
+
+    authorizations.forEach(
+        authorization -> {
+          final var request =
+              new CreateAuthorizationRequest(
+                  targetGroupId,
+                  AuthorizationOwnerType.GROUP,
+                  authorization.resourceKey(),
+                  convertResourceType(authorization.resourceType()),
+                  convertPermissions(authorization.permissions(), authorization.resourceType()));
+          try {
+            authorizationServices.createAuthorization(request).join();
+            logger.debug(
+                "Authorization created for group '{}' with permissions '{}'.",
+                targetGroupId,
+                request.permissionTypes());
+          } catch (final Exception e) {
+            if (!isConflictError(e)) {
+              throw new MigrationException(
+                  String.format(
+                      "Failed to create authorization for group '%s' with permissions '%s'",
+                      targetGroupId, request.permissionTypes()),
+                  e);
+            }
+            logger.debug(
+                "Authorization for group '{}' with permissions '{}' already exists, skipping creation.",
+                targetGroupId,
+                authorization.permissions());
+          }
+        });
+  }
+
+  private AuthorizationResourceType convertResourceType(final String resourceType) {
+    if (IDENTITY_PROCESS_DEFINITION_RESOURCE_TYPE.equalsIgnoreCase(resourceType)) {
+      return AuthorizationResourceType.PROCESS_DEFINITION;
+    }
+    if (IDENTITY_DECISION_DEFINITION_RESOURCE_TYPE.equalsIgnoreCase(resourceType)) {
+      return AuthorizationResourceType.DECISION_DEFINITION;
+    }
+    logger.debug("Unknown resource type: {}. Defaulting to UNSPECIFIED.", resourceType);
+    return AuthorizationResourceType.UNSPECIFIED;
+  }
+
+  private Set<PermissionType> convertPermissions(
+      final Set<String> permissions, final String resourceType) {
+    return switch (resourceType) {
+      case IDENTITY_DECISION_DEFINITION_RESOURCE_TYPE -> convertDecisionPermissions(permissions);
+      case IDENTITY_PROCESS_DEFINITION_RESOURCE_TYPE -> convertProcessPermissions(permissions);
+      default -> {
+        logger.warn("Unknown resource type: {}. Skipping permissions conversion.", resourceType);
+        yield Collections.emptySet();
+      }
+    };
```

Comment:
💭  style-wise we could have done both with either if-cascade or switch-case? let's align that?